pkg/execution/state/redis_state/queue.go

```diff
@@ -1546,50 +1686,34 @@ func (q *queue) Peek(ctx context.Context, partition *QueuePartition, until time.
 		}
 	}
 
-	if isPeekNext {
-		str, ok := items[0].(string)
-		if !ok {
-			return nil, fmt.Errorf("non-string value in next peek response: %T", items[0])
-		}
-
-		i, err := q.decodeQueueItemFromPeek(str, q.clock.Now())
-		if err != nil {
-			return nil, err
-		}
-		return []*osqueue.QueueItem{i}, nil
-	}
-
 	now := q.clock.Now()
 	return util.ParallelDecode(items, func(val any) (*osqueue.QueueItem, error) {
 		if val == nil {
-			q.logger.Error().Str("partition", partition.zsetKey(q.primaryQueueShard.RedisClient.kg)).Msg("nil item value in peek response")
+			q.logger.Error().Str("partition", opts.PartitionKey).Msg("nil item value in peek response")
 			return nil, nil
 		}
 
 		str, ok := val.(string)
 		if !ok {
 			return nil, fmt.Errorf("non-string value in peek response: %T", val)
 		}
-		return q.decodeQueueItemFromPeek(str, now)
-	})
-}
 
-func (q *queue) decodeQueueItemFromPeek(str string, now time.Time) (*osqueue.QueueItem, error) {
-	if str == "" {
-		return nil, fmt.Errorf("received empty string in decode queue item from peek")
-	}
+		if str == "" {
+			return nil, fmt.Errorf("received empty string in decode queue item from peek")
+		}
 
-	qi := &osqueue.QueueItem{}
-	if err := json.Unmarshal(unsafe.Slice(unsafe.StringData(str), len(str)), qi); err != nil {
-		return nil, fmt.Errorf("error unmarshalling peeked queue item: %w", err)
-	}
-	if qi.IsLeased(now) {
-		// Leased item, don't return.
-		return nil, nil
-	}
-	// The nested osqueue.Item never has an ID set;  always re-set it
-	qi.Data.JobID = &qi.ID
-	return qi, nil
+		qi := &osqueue.QueueItem{}
+		if err := json.Unmarshal(unsafe.Slice(unsafe.StringData(str), len(str)), qi); err != nil {
+			return nil, fmt.Errorf("error unmarshalling peeked queue item: %w", err)
+		}
+		if !opts.IgnoreLease && qi.IsLeased(now) {
```

Comment:
Just making sure, we skip leased (in progress) queue items? Though, if they're in the partition queue, they wouldn't be leased in the first place.

Reply:
there's a window that they're leased but also execution is paused because of the `migrate` flag in metadata.
if the lease didn't expire by then, then it'll need to move as well.